packages/node-resolve/src/resolveImportSpecifiers.js

```diff
@@ -37,7 +40,7 @@ function findExportKeyMatch(exportMap, subPath) {
   return null;
 }
 
-function mapSubPath(pkgJsonPath, subPath, key, value) {
+function mapSubPath(importer, pkgJsonPath, subPath, key, value) {
```

Comment:
```suggestion
function mapSubPath({ importer, pkgJsonPath, subPath, key, value }) {
```

The time has come to optionify the parameters for these functions imho (any parameter list greater than 3 args)

Reply:
Agreed

Reply:
Looking at this, these arguments appear to be similar in several functions. Maybe switching to objects can have advantages in that you might be able to just pass on entire objects without destructuring in some cases?

Reply:
unfortunately not, usually we change up some parameters